mechafil/power.py

```diff
@@ -1,13 +1,14 @@
 import numpy as np
 import pandas as pd
 import datetime
-from typing import Callable, Tuple
+from typing import Callable, Tuple, Union
+import numbers
 
 # --------------------------------------------------------------------------------------
 #  QA Multiplier functions
 # --------------------------------------------------------------------------------------
 def compute_qa_factor(
-    fil_plus_rate: float,
+    fil_plus_rate: Union[np.array, float],
     fil_plus_m: float = 10.0,
     duration_m: Callable = None,
     duration: int = None,
```

Comment:
If `fil_plus_rate` is an array, then the return of this function will also be an array. You need to update the return type of this function to `Union[np.array, float]`

Reply:
good catch. done.